railties/lib/rails/secrets.rb

```diff
@@ -30,20 +33,31 @@ def parse(paths, env:)
       end
 
       def generate_key
-        cipher = new_cipher
-        SecureRandom.hex(cipher.key_len)[0, cipher.key_len]
+        new_cipher.random_key.unpack("H*").first
       end
 
       def key
-        ENV["RAILS_MASTER_KEY"] || read_key_file || handle_missing_key
+        [(ENV["RAILS_MASTER_KEY"] || read_key_file || handle_missing_key)]
+          .pack("H*")
       end
 
-      def encrypt(text)
-        cipher(:encrypt, text)
+      def encrypt(data)
+        cipher = new_cipher.encrypt
+        cipher.key = key
+
+        iv = cipher.random_iv
+        ct = cipher.update(data) << cipher.final
+        tag = cipher.authenticated? ? cipher.auth_tag : ""
+
+        iv << tag << ct
       end
 
       def decrypt(data)
-        cipher(:decrypt, data)
+        cipher = new_cipher.decrypt
+        cipher.key = key
+        cipher.iv = data[0, cipher.iv_len]
```

Comment:
Should you be checking `cipher.authenticated?` here as well just in case anyone swaps out  `config.secrets_cipher`?